website/blog/2018-10-00-1.15.0.md

```diff
@@ -0,0 +1,1766 @@
+      : "hello world, branch a && !b"
+    : null;
+```
+
+#### Keep decorators inline if they were written inline ([#5188] by [@duailibe])
+
+In Prettier 1.14, we tried to solve an issue that decorators were always inlined,
+we thought inline decorators were rare cases so we could put them on their own line,
+but unfortunately it's not the case.
+Then we tried to find another better heuristic to identify when should we inline them,
+but we came up with the result after a long discussion in [#4924]
+that there's no a reliable to identify them,
+so we decided to respect the original source in Prettier 1.15.
+
+<!-- prettier-ignore -->
+```js
+// Input
+class Hello {
+  @decorator inline = 'value';
+
+  @decorator
+  ownLine = 'value';
+  
+  @decorator({
+    hello: 'world'
+  }) multiLine = 'value';
+}
+
+// Output (Prettier 1.14)
+class Hello {
+  @decorator
+  inline = "value";
+
+  @decorator
+  ownLine = "value";
+
+  @decorator({
+    hello: "world"
+  })
+  multiLine = "value";
+}
+
+// Output (Prettier 1.15)
+class Hello {
+  @decorator inline = "value";
+
+  @decorator
+  ownLine = "value";
+
+  @decorator({
+    hello: "world"
+  })
+  multiLine = "value";
+}
+```
+
+#### Respect original decorator order ([#5207] by [@duailibe])
+
+Decorators are still not standardized and
+[where should decorators on exported classes go] is not yet decided.
+To help proposal authors to get some feedbacks,
+Babel 7 added support for both decorators before and after the exported classes.
+Prettier 1.15 adds support for them and respects where did you put decorator.
+(Once the spec is standardized, we'll change it to be consistent and not respect user input.)
+
+<!-- prettier-ignore -->
+```js
+// decorator before export
+@decorator export class Bar {}
+
+// decorator after export
+export @decorator class Foo {}
+```
+
+[where should decorators on exported classes go]: https://babeljs.io/blog/2018/09/17/decorators#where-should-decorators-on-exported-classes-go
+
+#### Improved object-break heuristic ([#5205] by [@j-f1])
+
+Previously, Prettier would automatically break objects onto multiple lines if they didn’t fit within the print width.
+Prettier would also keep objects broken onto multiple lines if there was a newline somewhere inside them in the input code.
+This made it difficult to collapse an object since you had to manually merge the object onto one line.
+Since manual formatting changes are a task Prettier aims to eliminate, we’ve changed the behavior to only check for a newline between the `{` and the first key:
+
+<!-- prettier-ignore -->
+```js
+// Input
+const data = {  foo: 'bar',
+  baz: 'quux'
+}
+/* You’d get this format by deleting the newline after the `{` */
+
+// Output (Prettier 1.14)
+const data = {
+  foo: 'bar',
+  baz: 'quux'
+}
+
+// Output (Prettier 1.15)
+const data = { foo: 'bar', baz: 'quux' }
+
+```
+
+### JSX
+
+#### fix: splitting jsx text correctly ([#5006] by [@yuliaHope])
+
+TODO
```

Comment:
@lydell Can you help me with this item? as I don't have too much context on it.

Reply:
Ok!

Reply:
I've pushed a description. I hope I got it right, I'm in a bit of a hurry

Reply:
👍 Thanks!